File: springboot/security/basic-board_token/src/main/java/com/example/basicboard_token/controller/BoardController.java
package com.example.basicboard_token.controller;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class BoardController {

    @GetMapping("/")
    public String boardList() {
        return "board-list";
    }

    @GetMapping("/write")
    public String writeBoard() {
        return "board-write";
    }

    @GetMapping("/detail")
    public String detail(
            @RequestParam("id") Long id,
            Model model
    ) {
        model.addAttribute("id", id);
        return "board-detail";
    }

    @GetMapping("/update/{id}")
    public String update(
            @PathVariable("id") long id,
            Model model
    ) {
        model.addAttribute("id", id);
        return "board-update";
    }

    @GetMapping("/stats")
    public String stats() {
        return "board-stats";
    }
}

File: kotlin/kotlin-part-4/src/main/kotlin/org/example/kotlinpart4/controller/BoardController.kt
package org.example.kotlinpart4.controller

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestParam

@Controller
class BoardController {

    @GetMapping("/")
    fun boardList() = "board-list"

    @GetMapping("/write")
    fun write() = "board-write"

    @GetMapping("/detail")
    fun detail(
        @RequestParam("id") id: Long,
        model: Model
    ): String {
        model.addAttribute("id", id)
        return "board-detail"
    }

    @GetMapping("/update/{id}")
    fun update(@PathVariable("id") id: Long, model: Model): String {
        model.addAttribute("id", id)
        return "board-update"
    }

}